libraries/AP_Common/sorting.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdint.h>
+#include "sorting.h"
+
+/*
+  in-place insertion sort for small arrays of data. This is O(n) if
+  already sorted and O(n^2) for worst case (elements are reversed)
+  sort order is smallest first
+ */
+void insertion_sort_uint16(uint16_t *data, uint16_t n)
+{
+    for (uint16_t i=1; i<n; i++) {
+        uint16_t temp = data[i];
+        int16_t j = i - 1;
+
+        while (j >= 0 && data[j] > temp) {
+            data[j+1] = data[j];
+            j--;
+		}
+        data[j+1] = temp;
+    }
+}
+
+/*
+  remove duplicates from a sorted uint16_t array, returning the new
+  count
+ */
+uint16_t remove_duplicates_uint16(uint16_t *data, uint16_t n)
+{
+    uint16_t removed = 0;
```

Comment:
check is data is null or n==0?

Reply:
why? it just adds code when it falls through correctly without the check